compiler/qsc/src/interpret/tests.rs

```diff
@@ -712,6 +712,66 @@ mod given_interpreter {
             "#]].assert_eq(&res);
         }
 
+        #[test]
+        fn adaptive_qirgen() {
```

Comment:
Could you add a test case with an error? (Might help validate my other question about whether we need to wrap the errors with `WithSource`)

Reply:
Added test